main.go

```diff
@@ -1,7 +1,328 @@
 package main
 
-import "fmt"
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"time"
 
+	jwt "github.com/vladimiroff/jwt-go"
+
+	"github.com/gorilla/mux"
+)
+
+// Certificate Struct
+type Certificate struct {
+	ID        string    `json:"id"`
+	Title     string    `json:"title"`
+	CreatedAt string    `json:"createdAt"` // Will be a type date
+	OwnerID   string    `json:"ownerId"`
+	Year      int       `json:"year"`
+	Note      string    `json:"note"`
+	Transfer  *Transfer `json:"transfer"`
+}
+
+// User Struct
+type User struct {
+	ID    string `json:"id"`
+	Email string `json:"email"`
+	Name  string `json:"name"`
+}
+
+// Transfer Struct
+type Transfer struct {
+	To     string `json:"to"`
+	Status string `json:"status"`
+}
+
+// In memory data
+var certificates []Certificate              //Certificates
+var users []User                            //Users
+var mySigningKey = []byte("mockSigningKey") //mock signing key
```

Comment:
TODO will want to setup a signing key in the ENVIRONMENT variables, we do not want to commit in a signing key into a public repo. This is a private key.

Need  a good signing key.

https://www.cryptsoft.com/pkcs11doc/v220/group__SEC__12__22__3__GENERAL__LENGTH__SHA__256__HMAC.html

https://crypto.stackexchange.com/questions/34864/key-size-for-hmac-sha256 Topic about length